src/org/sosy_lab/java_smt/basicimpl/AbstractProver.java

```diff
@@ -22,8 +33,10 @@ public abstract class AbstractProver<T> implements BasicProverEnvironment<T> {
   protected final boolean generateUnsatCores;
   private final boolean generateUnsatCoresOverAssumptions;
   protected final boolean enableSL;
+  protected boolean closed = false;
 
   private final Set<Evaluator> evaluators = new LinkedHashSet<>();
+  private final List<Collection<BooleanFormula>> assertedFormulas = new ArrayList<>();
```

Comment:
Is there a benefit to `ArrayList` compared to the `Deque` that we use in most cases where an assertion stack is needed?